internal/private/private.go

```diff
@@ -81,8 +81,9 @@ type ImageDestination interface {
 // UploadedBlob is information about a blob written to a destination.
 // It is the subset of types.BlobInfo fields the transport is responsible for setting; all fields must be provided.
 type UploadedBlob struct {
-	Digest digest.Digest
-	Size   int64
+	Digest    digest.Digest
+	Size      int64
+	TOCDigest digest.Digest
```

Comment:
No transport is setting this?! And the transport wouldn’t know, after all.

I think if _anything_ should be managing this value:
- it can come from the source, if we don’t modify the blob
- it can be computed while compressing the blob to zstd:chunked, as a part of the copy
- it can be _discarded_ while decompressing a zstd:chunked blob, as part of the copy

so it initially seems to me that with `UploadedBlob`, the destination has no business providing/obtaining the value. (With `TryReuseBlob`, the case for the destination transport providing data is somewhat stronger.)

Especially if the digest ever became a part of the signature, we shouldn’t trust a `TOCDigest` value that some remote server, or something like that, claims, without validating it ourselves. Compare `canSubstituteBlobs`.

Reply:
Resolved in #2288.